charts/ocis/templates/idm/service.yaml

```diff
@@ -16,7 +16,9 @@ spec:
     - name: ldaps
       port: 9235
       protocol: TCP
+      appProtocol: tcp
```

Comment:
According to https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/ I would say `tls` is the right one here as this is also encrypted.

Reply:
Hmm, ok I was thinking about this but didn't document my rationale:

https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol links https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml , which does NOT mention `tls`. It does mention `ldaps` but I don't know if service meshes actually support / know about it or do a fallback to something like `tcp` anyways. For example Istio doesn't seem to know about it.

Therefore I thought the safest option for now is `tcp`. Also this is "only" the IDM which should be replaced in most cases anyways.

Reply:
Ok fine for me in this case.

Reply:
ugghh: https://cloud.google.com/kubernetes-engine/docs/how-to/secure-gateway

`The appProtocol field in the Service configuration only accepts capital letters for the protocol value (HTTP, HTTPS or HTTP2). Using lowercase letters results in using HTTP as the protocol with the backends.`

Reply:
I like it when there are contradictory standards.

Reply:
I would go with capital letters in this case, as Google and the k8s documentation have capital letters. Only istio has lowercase letters. I hope they can cope with capital letters as well.

Reply:
Ah wait. K8s also uses lowercase letters. Looked at the wrong line.

Reply:
Maybe for the sake of compatibility: 

Use something like 

```
{{ if .values.capitalAppProtocol }}{{ upper "http" }}{{ else }}{{ lower "http" }}
```

could be templated and then used as template by some like 

```
{{- include "ocis.rightAppProtocol" "http" -}}
```